### PR TITLE
Added new options to Evapotranspiration dialog

### PR DESCRIPTION
**Pull request by MeSophie**
I made some modifications on this Dialog according to @rdstern comment on PR#7611.
@Patowhiz, @N-thony , @AmsaleEjigu please can you have a look.

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@MeSophie that's a good start.  Could you also look at the other suggestions I made in the layout in #7611.
Then, in addition, what I haven't done is to make much use of the dialogue.  I suggest you also do this with the ICRISAT data and with other data sets.  And that could be in parallel with the cosmetic layout improvements.
That should make it easy to write the help, and possibly produce a video on the use of the dialogue.
To start I wonder what examples are supplied with the package itself.  Then there is the bohicon data in the Instat library climatic guide data sets (omit the top about 13 lines).  And you should download the ICRISAT data I mention in 7611.

Those examples will (I hope) be fairly trivial to run, but let's see.  

Then we need to see how to use it when there are missing values and when elements are missing.  I think the function will cope, but this will need investigation.  I suggest it will be efficient for you to start with the documentation from the package and we may eventually have to look at the function itself - though I hope now!

**Comment by MeSophie** [comment]
@rdstern Please can you go and look the comment that I made on PR#7611. Thanks

**Comment by MeSophie** [comment]
I noticed that after filling in the 4 main parameters, the **Ok** button is active but if we delete any of these parameters, the **Ok** button is still active, so I have corrected this problem.

image

**Review by rdstern — CHANGES_REQUESTED** [review]
@MeSophie I am now c learer on the possible improvements in layout.
image

In the image above
a) Change the label Radiation to Solar.
b) In the pull-down, change Data to Radiation
c) Don't have the Wind in a block.  Instead just have the existing Checkbox with label Wind Speed, directly underneath the Solar label.  Default - as now, is checked and the receiver is there.  If unchecked, then - as now - the receiver is no longer visible.
d) Please check whether the options for missing are all the same for the second main radio-button.  If so, then just improve the spacing a bit.

I then tried with the Bohicon data - here it is.  It seems to work!
[bohicon.zip](https://github.com/africanmathsinitiative/R-Instat/files/10058258/bohicon.zip)

Finally I noticed that the humidity data is not yet in our defined climatic data, so that's a new issue.  But let's see if this one can be merged first - and then we are ready to test it!

**Comment by MeSophie** [comment]
> Please check whether the options for missing are all the same for the second main radio-button. If so, then just improve the spaci

@rdstern  The missing options contained in the missing options sub-dialogue are common to both radio buttons. I change the Radiation label to Solar but is still Radiation. So @Vitalis95  will change also the database.

**Comment by rdstern** [comment]
@MeSophie thank you for the explanation about Solar.
There is now one further (welcome) change I am afraid!
image

The change is in the Define Climatic data, where the humidity variables are now defined - and this means that they can all be included automatically when the dialogue is loaded.  And that will make the method simpler to use.

So can you check how the existing elements are added automatically into the dialogue, and extend that now to the humidity variables.  You may have to wait till the Define Climatic Data dialogue is merged.

**Comment by MeSophie** [comment]
> @MeSophie thank you for the explanation about Solar. There is now one further (welcome) change I am afraid! image
> 
> The change is in the Define Climatic data, where the humidity variables are now defined - and this means that they can all be included automatically when the dialogue is loaded. And that will make the method simpler to use.
> 
> So can you check how the existing elements are added automatically into the dialogue, and extend that now to the humidity variables. You may have to wait till the Define Climatic Data dialogue is merged.

@rdstern I'm working on it. All climatic data are include automatically for  Bihicon data but for Patancheru there is small issue with radiation.

**Comment by MeSophie** [comment]
@lloyddewit , @rdstern this PR is ready to review.
There is still the radiation Solar parameter which does't fill automatically but @N-thony suggest that after merge the PR we can create and issue for this.

**Review by lloyddewit — APPROVED** [review]

**Review by rdstern — APPROVED** [review]
GReat

**Comment by lloyddewit** [comment]
@rdstern if you approve, then I think we can merge, thanks

@ChrisMarsh82 The compile checks are failing (see screenshot below). However, they are failing on the compilation error that was fixed in PR #7997 . Master now builds. I don't understand why this PR tries to merge/build against a previous master commit. I reran the GitHub build checks in 4 other PRs and they all passed so I don't believe there is a general problem. I propose to merge this PR anyway. What's your view?

image

**Comment by lloyddewit** [comment]
>  I reran the GitHub build checks in 4 other PRs and they all passed so I don't believe there is a general problem.

@ChrisMarsh82 
I just found that the build checks also fail in PR #7930 for the same reason. Not sure why.

File: instat/sdgMissingOptionsEvapotranspiration.vb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿' R- Instat
+' Copyright (C) 2015-2017
+'
+' This program is free software: you can redistribute it and/or modify
+' it under the terms of the GNU General Public License as published by
+' the Free Software Foundation, either version 3 of the License, or
+' (at your option) any later version.
```
Comment (lloyddewit) [inline instat/sdgMissingOptionsEvapotranspiration.vb:7]: just a test, please ignore
```suggestion
' (at your option) any later version. 
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by lloyddewit** [comment]
@ChrisMarsh82 Your theory was correct! thank you!
Checks seem OK now. I'll merge this PR tomorrow.

**Review by rdstern — APPROVED** [review]
@MeSophie seems to work fine - even adding the humidity!  But not the sunshine - so that can become an issue.  Perhaps together with the 3rd method to add!

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**